src/wp-includes/class-wp-comment-query.php

```diff
@@ -802,9 +802,18 @@ protected function get_comment_ids() {
 
 		// Falsey search strings are ignored.
 		if ( strlen( $this->query_vars['search'] ) ) {
+			/**
+			 * Specifying allowed fields directly, excluding comment_author_email field for unlogged users, to avoid unwanted data exposure for public endpoint.
+			 * Leaving possibility to search by email if function called internally for better backward compatibility.
+			 * @link https://core.trac.wordpress.org/ticket/53784
+			 */
+			$allowed_columns = array( 'comment_author', 'comment_author_url', 'comment_author_IP', 'comment_content' );
```

Comment:
We should probably be moving the `comment_author_IP` field as well.

Reply:
cd15116e8a5933a64a8d3d98517da8b0f0c876ac